### ISSUE TITLE
[java] Private classes and members are flagged for not having comments

### ISSUE THREAD

**Issue by espertus** [pmd/pmd#4570]
**Affects PMD Version:** 

6.55.0

**Rule:**
CommentRequired

<https://docs.pmd-code.org/latest/pmd_rules_java_documentation.html#commentrequired>

**Description:**
Private classes and members are flagged for not having comments. While I could disable the rule entirely, I would prefer to be able to disable it just for `private` classes and members and keep it for other visibility levels.

I see that SonarPMD has an [unmerged PR](https://github.com/jborgers/sonar-pmd/pull/31/files) about this issue.

**Code Sample demonstrating the issue:**

```java
public class Test {
    private int secret;
}
```

**Expected outcome:**

PMD reports a violation at line 2, which feels like a false positive to me, since I don't think private members should require comments.

**Running PMD through:** *IntelliJ plugin*

**Comment by jsotuyod** [comment]
Yeah, that's very reasonable actually.

We already have `publicMethodCommentRequirement` and `protectedMethodCommentRequirement` which go in this direction, but only for methods, and only considers these 2 visibilities.

I don't think doing the *n x m* element-visibility pairs is sustainable.

I can think a reasonable path forward would be:
1. We have a property for minimum visibility (valid values being -in order- 'private', 'package', 'protected', 'public')
2. We have a property for each field type with the current 'Required', 'Ignored', 'Unwanted' values
  1. We deprecate both `publicMethodCommentRequirement` and `protectedMethodCommentRequirement` in favor of a single `methodCommentRequirement`.
  2. Accessors and overrides are still treated separately (I think it makes sense to have a separate config for those)

We wouldn't be able to express things such as "all public methods and protected + public fields", but IMHO it doesn't make sense… if you want to document for extension, why limit it to fields but not methods?

**Comment by mstup** [comment]
Be able to turn on/off requirement to commenting:
- classes/interfaces/enum/records - public and protected (maybe default)
- methods - public and protected (maybe default)
- fields - public and protected

I would rather use property like this instead of bunch what do we have now. This is exactly how code is commented for use inside of company but not as a framework where each should have this possibility to understand all your code completely.
